perception/object_recognition/tracking/multi_object_tracker/CMakeLists.txt

```diff
@@ -1,34 +1,29 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(multi_object_tracker)
 
-add_compile_options(-std=c++14)
+### Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic -Werror)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  autoware_perception_msgs
-  roscpp
-  tf2
-  tf2_ros
-  unique_id
-  uuid_msgs
-)
+### Find Packages
+find_package(ament_cmake_auto REQUIRED)
 
-find_package(Eigen3 REQUIRED)
+## Find Eigen Dependencies
+find_package(eigen3_cmake_module REQUIRED)
```

Comment:
When I remove the `find_package(eigen_cmake_module)` and the `find_package(Eigen3)` calls, I cannot find the Eigen headers. I call the `ament_auto_find_build_dependencies()` call after these because I thought it grouped the calls a little better, though I see how it looks quite confusing I'll change this.

Are there other packages which have used the Eigen libraries and could just depend on the `ament_auto_find_build_dependencies()` call to find the package?

Reply:
@jilaada I used it [here](https://github.com/tier4/Pilot.Auto/blob/ros2/common/math/osqp_interface/CMakeLists.txt) w/o issues. Could update that to use `Eigen3::Eigen` as @nnmm recommended but otherwise it's equivalent to your context here

Reply:
Thanks for the information. I've changed it according to @nnmm suggestion in this comment: https://github.com/tier4/Pilot.Auto/pull/24#discussion_r512511753. Works excellently!

Reply:
You're probably aware of this yourself, but just a side note: I wouldn't go too far with "if it works and pulls in the least number of dependencies, it's correct" – I often noticed that colcon does not complain if you leave out theoretically-necessary dependencies in package.xml and CMakeLists.txt. Those packages are then subtly broken and only work by accident, or will reveal problems in packages that depend on them.

So even if your package would work only with `ament_auto_find_build_dependencies()`, I'd think the way you did it here is correct since [I wouldn't assume `ament_auto_find_build_dependencies()` will `find_package()` in any particular order](https://github.com/ament/ament_cmake/blob/f123edf40390be5bcfa015105c76ccd763acac3d/ament_cmake_auto/cmake/ament_auto_find_build_dependencies.cmake#L62) and [the `eigen3_cmake_module` readme says to find `Eigen3` after `eigen3_cmake_module`](https://github.com/ros2/eigen3_cmake_module#edit-your-cmakeliststxt).

Reply:
The second doc you linked was what I used as reference to add the eigen dependencies. I thought the order in which packages are added doesn't matter - but now that I know I'll keep that in mind for the future. I'll add this find_package call back. Added back here: 160b275